src/parser/expression.js

```diff
@@ -62,15 +63,18 @@ const expression = (
       getAssociativty(previous) === "left"
     ) {
       if (value === "," && previous.type === Syntax.FunctionCall) {
-break;
-}
+        break;
+      }
       // if (value === ":" && previous.type === Syntax.Pair) break;
       consume();
     }
   };
 
   const processPunctuator = () => {
     switch (ctx.token.value) {
+      case ".":
+        operators.push(ctx.token);
```

Comment:
Why are you handling the dot separately? It should be parsed as part of the `default` case for operators.